### ISSUE TITLE
Crash during flight on pressing left Shift key

### ISSUE THREAD

**Issue by pirogronian** [TheOpenSpaceProgram/osp-magnum#67]
Version: fresh (commit ad0de758d6007c23a082f62ab2ed83657639e817)
Recreation: run, type "flight", type 'v' (to focus on first ship), then left shift. Edit: similar crash happens if I focus on second ship and type 'z' (also tested 'x' and 'c').
Console output:
`osp-magnum: /home/ja/Public/Programs/osp-magnum/src/../3rdparty/entt/src/entt/entity/registry.hpp:991: auto entt::basic_registry< <template-parameter-1-1> >::try_get(entt::basic_registry< <template-parameter-1-1> >::entity_type) [with Component = {adera::active::machines::MachineContainer}; Entity = osp::active::ActiveEnt; entt::basic_registry< <template-parameter-1-1> >::entity_type = osp::active::ActiveEnt]: Assertion `valid(entity)' failed.`
Backtrace:
```
#0  0x00007ffff7441ef5 in raise () from /usr/lib/libc.so.6
#1  0x00007ffff742b862 in abort () from /usr/lib/libc.so.6
#2  0x00007ffff742b747 in __assert_fail_base.cold () from /usr/lib/libc.so.6
#3  0x00007ffff743a646 in __assert_fail () from /usr/lib/libc.so.6
#4  0x000055555557ce89 in entt::basic_registry<osp::active::ActiveEnt>::try_get<adera::active::machines::MachineContainer> (
    this=0x7ffff00f9e50, entity=(unknown: 0xfffff))
    at /home/ja/Public/Programs/osp-magnum/src/../3rdparty/entt/src/entt/entity/registry.hpp:991
#5  0x000055555557cecc in osp::active::ActiveScene::reg_try_get<adera::active::machines::MachineContainer> (
    this=0x7ffff00f9e40, ent=(unknown: 0xfffff)) at /home/ja/Public/Programs/osp-magnum/src/./osp/Active/ActiveScene.h:141
#6  0x000055555557a7ae in adera::active::machines::SysMachineRocket::update_physics (this=0x7ffff021fd90, rScene=...)
    at /home/ja/Public/Programs/osp-magnum/src/adera/Machines/Rocket.cpp:128
#7  0x000055555557a374 in operator() (__closure=0x7ffff021ff70, rScene=...)
    at /home/ja/Public/Programs/osp-magnum/src/adera/Machines/Rocket.cpp:84
#8  0x000055555557be43 in std::__invoke_impl<void, adera::active::machines::SysMachineRocket::SysMachineRocket(osp::active::ActiveScene&)::<lambda(osp::active::ActiveScene&)>&, osp::active::ActiveScene&>(std::__invoke_other, struct {...} &) (
    __f=...) at /usr/include/c++/10.2.0/bits/invoke.h:60
#9  0x000055555557bd18 in std::__invoke_r<void, adera::active::machines::SysMachineRocket::SysMachineRocket(osp::active::ActiveScene&)::<lambda(osp::active::ActiveScene&)>&, osp::active::ActiveScene&>(struct {...} &) (__fn=...)
    at /usr/include/c++/10.2.0/bits/invoke.h:110
#10 0x000055555557bbef in std::_Function_handler<void(osp::active::ActiveScene&), adera::active::machines::SysMachineRocket::SysMachineRocket(osp::active::ActiveScene&)::<lambda(osp::active::ActiveScene&)> >::_M_invoke(const std::_Any_data &, osp::active::ActiveScene &) (__functor=..., __args#0=...) at /usr/include/c++/10.2.0/bits/std_function.h:291
#11 0x00005555555a7b23 in std::function<void (osp::active::ActiveScene&)>::operator()(osp::active::ActiveScene&) const (
    this=0x7ffff021ff70, __args#0=...) at /usr/include/c++/10.2.0/bits/std_function.h:622
#12 0x00005555555a5b59 in osp::FunctionOrder<void (osp::active::ActiveScene&)>::call<osp::active::ActiveScene&>(osp::active::ActiveScene&) (this=0x7ffff00f9ed0) at /home/ja/Public/Programs/osp-magnum/src/osp/Active/../FunctonOrder.h:154
#13 0x00005555555a35ae in osp::active::ActiveScene::update (this=0x7ffff00f9e40)
    at /home/ja/Public/Programs/osp-magnum/src/osp/Active/ActiveScene.cpp:196
#14 0x0000555555645258 in testapp::OSPMagnum::drawEvent (this=0x7ffff0000b60)
    at /home/ja/Public/Programs/osp-magnum/src/test_application/OSPMagnum.cpp:86
#15 0x0000555555666ed1 in Magnum::Platform::Sdl2Application::mainLoopIteration (this=0x7ffff0000b60)
    at /home/ja/Public/Programs/osp-magnum/3rdparty/magnum/src/Magnum/Platform/Sdl2Application.cpp:974
#16 0x0000555555666450 in Magnum::Platform::Sdl2Application::exec (this=0x7ffff0000b60)
    at /home/ja/Public/Programs/osp-magnum/3rdparty/magnum/src/Magnum/Platform/Sdl2Application.cpp:802
#17 0x00005555556495da in testapp::test_flight (pMagnumApp=std::unique_ptr<testapp::OSPMagnum> = {...}, rOspApp=..., 
    args=...) at /home/ja/Public/Programs/osp-magnum/src/test_application/flight.cpp:154
#18 0x000055555565bf04 in std::__invoke_impl<void, void (*)(std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> >&, osp::OSPApplication&, Magnum::Platform::Sdl2Application::Arguments), std::reference_wrapper<std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> > >, std::reference_wrapper<osp::OSPApplication>, Magnum::Platform::Sdl2Application::Arguments> (
    __f=@0x555555a0f978: 0x555555649191 <testapp::test_flight(std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> >&, osp::OSPApplication&, Magnum::Platform::Sdl2Application::Arguments)>)
    at /usr/include/c++/10.2.0/bits/invoke.h:60
#19 0x000055555565bdce in std::__invoke<void (*)(std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> >&, osp::OSPApplication&, Magnum::Platform::Sdl2Application::Arguments), std::reference_wrapper<std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> > >, std::reference_wrapper<osp::OSPApplication>, Magnum::Platform::Sdl2Application::Arguments> (
    __fn=@0x555555a0f978: 0x555555649191 <testapp::test_flight(std::unique_ptr<testapp::OSPMagnum, std::default_delete<testap--Type <RET> for more, q to quit, c to continue without paging--c
p::OSPMagnum> >&, osp::OSPApplication&, Magnum::Platform::Sdl2Application::Arguments)>) at /usr/include/c++/10.2.0/bits/invoke.h:95
#20 0x000055555565bcc5 in std::thread::_Invoker<std::tuple<void (*)(std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> >&, osp::OSPApplication&, Magnum::Platform::Sdl2Application::Arguments), std::reference_wrapper<std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> > >, std::reference_wrapper<osp::OSPApplication>, Magnum::Platform::Sdl2Application::Arguments> >::_M_invoke<0ul, 1ul, 2ul, 3ul> (this=0x555555a0f958) at /usr/include/c++/10.2.0/thread:264
#21 0x000055555565bc46 in std::thread::_Invoker<std::tuple<void (*)(std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> >&, osp::OSPApplication&, Magnum::Platform::Sdl2Application::Arguments), std::reference_wrapper<std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> > >, std::reference_wrapper<osp::OSPApplication>, Magnum::Platform::Sdl2Application::Arguments> >::operator() (this=0x555555a0f958) at /usr/include/c++/10.2.0/thread:271
#22 0x000055555565bc2a in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> >&, osp::OSPApplication&, Magnum::Platform::Sdl2Application::Arguments), std::reference_wrapper<std::unique_ptr<testapp::OSPMagnum, std::default_delete<testapp::OSPMagnum> > >, std::reference_wrapper<osp::OSPApplication>, Magnum::Platform::Sdl2Application::Arguments> > >::_M_run (this=0x555555a0f950) at /usr/include/c++/10.2.0/thread:215
#23 0x00007ffff7802bc4 in std::execute_native_thread_routine (__p=0x555555a0f950) at /build/gcc/src/gcc/libstdc++-v3/src/c++11/thread.cc:80
#24 0x00007ffff7f8d299 in start_thread () from /usr/lib/libpthread.so.0
#25 0x00007ffff7504153 in clone () from /usr/lib/libc.so.6
```

**Comment by Capital-Asterisk** [comment]
Yep, this is kind of a known issue. The spamcan part does not have a fuel tank, and the null checks aren't done properly. PR #65 fixes and break this with a similar issue. I'll leave some comments on z-adam's PR.

**Comment by Capital-Asterisk** [comment]
Fixed in #65
